java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -845,6 +874,35 @@ private void buildNewClass(NewClassTree tree) {
     build(Lists.reverse(tree.arguments()));
   }
 
+  private void handleExceptionalPaths(Symbol symbol) {
+    TryStatement pop = enclosingTry.pop();
+    TryStatement tryStatement;
+    if(!enclosedByCatch.peek()) {
+      tryStatement = pop;
+    } else {
+      tryStatement = enclosingTry.peek();
+    }
+    enclosingTry.push(pop);
+    if(pop != outerTry) {
+      currentBlock = createBlock(currentBlock);
+      currentBlock.exceptions.add(exitBlocks.peek());
+    }
+    if (symbol.isMethodSymbol()) {
+      List<Type> thrownTypes = ((Symbol.MethodSymbol) symbol).thrownTypes();
+      if (!thrownTypes.isEmpty()) {
```

Comment:
no need to check if the thrownTypes list is empty, as the `forEach` will do nothing if it's the case.